source/AttributeStore.h

```diff
@@ -0,0 +1,233 @@
+/* AttributeStore.h
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ATTRIBUTE_STORE_H_
+#define ATTRIBUTE_STORE_H_
+
+#include "attribute/Attribute.h"
+#include "attribute/AttributeAccess.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Dictionary.h"
+
+#include <cmath>
+#include <functional>
+#include <limits>
+#include <map>
+#include <string>
+
+class AttributeStore {
+public:
+	// Checks whether the specified attribute is defined here.
```

Comment:
IsPresent only seems to be used in one location where `"required crew"` gets set based on `"automaton"`. I suggest to remove all IsPresent support and just use Get() for `"automaton"` (and check if it returns zero or non-zero).

Reply:
Yeah, that makes sense. That's generally already what happens in the background.